test/benchmark/iam/LanguageFeatureTest.java

```diff
@@ -0,0 +1,179 @@
+ *
+ */
+
+package com.vaticle.typedb.core.reasoner.benchmark.iam;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class LanguageFeatureTest{
+
+    private static final String database = "iam-benchmark-language-features";
+    private static final BenchmarkRunner benchmarker = new BenchmarkRunner(database);
+
+    public LanguageFeatureTest() { }
+
+    @BeforeClass
+    public static void setup() throws IOException {
+        benchmarker.setUp();
+        benchmarker.loadSchema("schema_types.tql");
+        benchmarker.loadSchema("schema_rules_optimised.tql");
+        benchmarker.loadSchema("schema_rules_test_specific.tql");
+        benchmarker.loadData("data.typedb");
+        benchmarker.warmUp();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        benchmarker.tearDown();
+    }
+
+    @After
+    public void reset() {
+        benchmarker.reset();
+    }
+
+    @Test
+    public void testOwnership() {
+        String query = "match\n" +
+                "$x has parent-company-name $n;\n";
+        Benchmark benchmark = new Benchmark("ownership", query, 0); // TODO: Data only exists in the full data-set
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(100);
+        benchmark.assertCounters(50, 0, 2, 2);
+    }
+
+    @Test
+    public void testAttributeWithoutOwnership() {
+        String query = "match\n" +
+                "$n isa parent-company-name;\n";
+        Benchmark benchmark = new Benchmark("attribute-without-ownership", query, 0); // TODO: Data only exists in the full data-set
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(100);
+        benchmark.assertCounters(50, 0, 1, 1);
+    }
+
+    @Test
+    public void testRelationWithRolePlayers() {
+        String query = "match\n" +
+                "$ac (object: $obj, action: $action) isa access; \n" +
+                "$obj has id \"root/engineering/typedb/src\";" +
+                "$action has name \"delete file\";\n" +
+                "$p (subject: $s, access: $ac) isa permission;\n";
+        Benchmark benchmark = new Benchmark("relation-with-role-players", query, 10);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(1000);
+        benchmark.assertCounters(500, 22, 30, 72);
+    }
+
+    @Test
+    public void testRelationWithoutRolePlayer() {
+        String query = "match\n" +
+                "$p isa permission;\n";
+        Benchmark benchmark = new Benchmark("relation-without-role-players", query, 2783);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(5000);
```

Comment:
this is already a pretty interesting test from the baseline! This means we're doing approximately 2700 answers in 5 seconds, or about 1 answer every 2 milliseconds. For simple inference I'd hope we can get this to 10x faster in Rust.

Reply:
These are the kinds of high-level pointers that is exactly the point of a simple inference test (to address your push back on "whats the point" of these tests).

Reply:
This isn't actually a simple baseline test.
Computing permissions recursively computes 3 transitive relations and is the difficult part of the iam-dataset.
There's no simple relation to infer in the IAM dataset. 
My question was never, "What's the point of simple tests?". It was: "If we do want baseline tests, why not add them in the existing reasoner benchmark?" as opposed to trying to awkwardly fit them into the iam dataset